File: _BHCVRP_Python_Version/BHCVRP_Python_Version/cujae_inf_citi_om/exceptions/DistanceNotAccessibleException.py
class DistanceNotAccessibleException(Exception):
    def __init__(self, distance=None):
        if distance is not None:
            self.distance = distance
            
    def __init__(self, message=None):
        if message is not None:
            super().__init__(message)

    def get_distance(self):
        return getattr(self, 'distance', None)

File: BHCVRP_Java_Version/src/cujae/inf/citi/om/exceptions/DistanceNotAccessibleException.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package cujae.inf.citi.om.exceptions;

/**
 *
 * @author nanda
 */

//Clase que maneja la excepción lanzada cuando la distancia no es accesible.

public class DistanceNotAccessibleException extends Exception{
    private float distance;
    
    public DistanceNotAccessibleException(float distance){
        this.distance = distance;
    
    }
    
    public float getDistance(){
        return distance;
    }
    
    public DistanceNotAccessibleException(String message){   
        super(message);
    }
}

